tools/cli/commands/changelog.js

```diff
@@ -290,80 +289,102 @@ async function checkSpecialProjects( needChangelog ) {
  * @param {argv} argv - the arguments passed.
  */
 async function changelogAdd( argv ) {
-	if ( argv._[ 1 ] === 'add' && ! argv.project ) {
-		const needChangelog = await checkChangelogFiles();
-		const uniqueProjects = await checkSpecialProjects( needChangelog );
+	let needChangelog;
+	const defaultProjects = [];
+	const uniqueProjects = [];
+	const defaultTypes = {
+		security: 'Improves or modifies the security of the project.',
+		added: 'Added new functionality.',
+		changed: 'Changed existing functionality.',
+		deprecated: 'Deprecated existing functionality.',
+		removed: 'Removed existing functionality.',
+		fixed: 'Fixed a bug.',
+	};
+
+	if ( argv.project ) {
+		needChangelog = [ argv.project ];
+	}
 
-		// If we don't detect any modified projects, shortcircuit to default changelogger.
-		if ( needChangelog.length === 0 && uniqueProjects.length === 0 ) {
+	// If we weren't passed a project, check if any projects need changelogs.
+	if ( argv._[ 1 ] === 'add' && ! argv.project ) {
+		needChangelog = await checkChangelogFiles();
+		if ( needChangelog.length === 0 ) {
 			console.log(
 				chalk.green(
-					'Did not detect a touched project that still need a changelog. You can still add a changelog manually.'
+					'Did not detect a touched project that still needs a changelog. You can still add a changelog manually by running `jetpack changelog add <type/project>'
 				)
 			);
-			changelogArgs( argv );
 			return;
 		}
+	}
 
-		const promptType = await changelogAddPrompt( argv, needChangelog, uniqueProjects );
+	const projectChangeTypes = await getProjectChangeTypes( needChangelog );
 
-		// Bail if user doesn't want to auto-add.
-		if ( ! promptType.autoAdd && ! promptType.autoPrompt ) {
-			console.log(
-				chalk.green(
-					`Auto changelog cancelled. You can run 'jetpack changelog add [project-type/project]' to add changelogs individually.`
-				)
-			);
-			return;
+	// Iterate over the project types, and if it's not default, add it to uniqueProjects.
+	for ( const proj in projectChangeTypes ) {
+		if ( projectChangeTypes[ proj ] !== 'default' ) {
+			uniqueProjects.push( proj );
+		} else {
+			defaultProjects.push( proj );
+			projectChangeTypes[ proj ] = defaultTypes;
 		}
+	}
+
+	// Confirm what projects we're adding a changelog to, and how we want to add them.
+	const promptConfirm = await changelogAddPrompt( argv, defaultProjects, uniqueProjects );
```

Comment:
We need to avoid calling `changelogAddPrompt` if `argv.project` is set. Or `changelogAddPrompt` should just assume "yes" if `argv.project` is set, either way works.

Reply:
Right now it still confirms the project we're releasing a changelog file to on line 828 - you think we should skip the confirmation altogether since someone is passing the project manually?

Reply:
Yes. That's the current behavior in trunk, and adding the extra confirmation seems silly.